staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -158,11 +158,16 @@ func diffMaps(original, modified map[string]interface{}, t reflect.Type, ignoreC
 		switch originalValueTyped := originalValue.(type) {
 		case map[string]interface{}:
 			modifiedValueTyped := modifiedValue.(map[string]interface{})
-			fieldType, _, _, err := forkedjson.LookupPatchMetadata(t, key)
+			fieldType, fieldPatchStrategy, _, err := forkedjson.LookupPatchMetadata(t, key)
 			if err != nil {
 				return nil, err
 			}
 
+			if fieldPatchStrategy == replaceDirective {
```

Comment:
After short investigation it looks like we should be good with the current approach of always replacing. The `ignoreChangesAndAdditions` and `ignoreDeletions` flags are used only internal in 3-way merge, which with our replace approach should be a safe call.